test/e2e/client.go

```diff
@@ -26,19 +26,24 @@ const (
 	// DefaultPollInterval is the default interval for polling
 	DefaultPollInterval = 5 * time.Second
 	// DefaultPollTimeout is the default timeout for polling
-	DefaultPollTimeout = 5 * time.Minute
+	DefaultPollTimeout = 10 * time.Minute
 )
 
+type ClientOptions struct {
+	conditionCheckTimeout time.Duration
+}
+
 type Client struct {
-	kClient        client.Client
-	timeout        time.Duration
-	ctx            context.Context
-	kubeconfigPath string
+	kClient               client.Client
+	timeout               time.Duration
+	ctx                   context.Context
+	kubeconfigPath        string
+	conditionCheckTimeout time.Duration
 }
 
 var singletonClient *Client
 
-func GetClient() (*Client, error) {
+func GetClient(options *ClientOptions) (*Client, error) {
```

Comment:
so even if someone wants to change options, it won't be reflected, when client (singleton) exists already?